extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaTuningConfig.java

```diff
@@ -241,8 +269,15 @@ public boolean equals(Object o)
         : that.basePersistDirectory != null) {
       return false;
     }
-    return indexSpec != null ? indexSpec.equals(that.indexSpec) : that.indexSpec == null;
-
+    if (indexSpec != null ? !indexSpec.equals(that.indexSpec) : that.indexSpec != null) {
```

Comment:
You can generate the forms which that I suggested with intelliJ as well, need to choose another option in generation dialog. Once you choose it, it will be the default later.